packages/slate-editor/src/modules/editor-v4-floating-add-menu/lib/shouldShowMenuButton.ts

```diff
@@ -17,6 +17,9 @@ export function shouldShowMenuButton(editor: Editor): boolean {
     }
 
     const text = Node.string(currentNode);
+    const hasNoPlaceholder = currentNode.children.every(
+        (child) => !('type' in child && child.type === PLACEHOLDER_NODE_TYPE),
```

Comment:
I think it is better to use Node.nodes(currentNode) to get children and use isPlaceholderNode to detect if a node is a placeholder

Also i think it is better to check that the first node is not a placeholder, not that there are some placeholders in the text

Reply:
You can use  `Array.from(Node.nodes(currentNode))`
> For some reason, the first node is empty text, so I can't be sure in which position it can be.

As i understand the issue is when the placeholder is first node in paragraph, if you would have space and the placeholder as i understand there will be no issues

Reply:
So, just check always second place for placeholder?

Reply:
Ok, i got what you mean, i think you can leave checking for all children. But i would reverse logic, i would check if line has placeholders and used currentNode.children.some

Reply:
done